WordPress/src/main/java/org/wordpress/android/ui/main/jetpack/migration/JetpackMigrationViewModel.kt

```diff
@@ -9,42 +9,55 @@ import org.wordpress.android.fluxc.store.AccountStore
 import org.wordpress.android.fluxc.store.SiteStore
 import org.wordpress.android.ui.main.jetpack.migration.JetpackMigrationViewModel.ActionButton.WelcomePrimaryButton
 import org.wordpress.android.ui.main.jetpack.migration.JetpackMigrationViewModel.ActionButton.WelcomeSecondaryButton
-import org.wordpress.android.ui.main.jetpack.migration.JetpackMigrationViewModel.StepUiState.Welcome
+import org.wordpress.android.ui.main.jetpack.migration.JetpackMigrationViewModel.UiState.Content
+import org.wordpress.android.ui.main.jetpack.migration.JetpackMigrationViewModel.UiState.Loading
 import org.wordpress.android.ui.utils.UiString
 import org.wordpress.android.ui.utils.UiString.UiStringRes
 import org.wordpress.android.util.GravatarUtilsWrapper
 import org.wordpress.android.util.SiteUtilsWrapper
 import javax.inject.Inject
 
-const val SITE_ICON_SIZE = 60 * 3
-const val USER_AVATAR_SIZE = 32 * 3
-
 @HiltViewModel
 class JetpackMigrationViewModel @Inject constructor(
     private val siteStore: SiteStore,
     private val accountStore: AccountStore,
     private val siteUtilsWrapper: SiteUtilsWrapper,
     private val gravatarUtilsWrapper: GravatarUtilsWrapper,
 ) : ViewModel() {
-    private val _uiState = MutableStateFlow(UiState())
+    private val _uiState = MutableStateFlow<UiState>(Loading)
     val uiState: StateFlow<UiState> = _uiState
 
+    // TODO Review this after data sync work is done
     fun onAccountInfoLoaded() {
-        _uiState.value = _uiState.value.copy(userAvatarUrl = getAvatarUrl())
+        if (isDataAvailable()) {
+            initWelcomeState()
+        }
     }
 
+    // TODO Review this after data sync work is done
     fun onSiteListLoaded() {
-        _uiState.value = _uiState.value.copy(
-                stepState = Welcome(
-                        sites = getSiteList(),
-                        primaryActionButton = WelcomePrimaryButton(::onContinueClicked),
-                        secondaryActionButton = WelcomeSecondaryButton(::onHelpClicked),
-                ),
+        if (isDataAvailable()) {
+            initWelcomeState()
+        }
+    }
+
+    private fun isDataAvailable() = accountStore.account.userName.isNotEmpty() && siteStore.sites.isNotEmpty()
```

Comment:
I'm not sure these checks are valid. Apparently there's an `account` object in the store even before the user account is fetched, and it has default values.

I noticed the check on `userName` not empty was used in other parts of the code, but the site list could in theory be empty in some cases even after the sites data is fetched.

Given this is a temporary logic, and it depends on the updates to the data layer, a solution might also be to set viewModel boolean flags in the events callback and initialize the Welcome Screen data when both are true.

But that's also not the most elegant solution 🤷 .

Reply:
Indeed, these developments will need to link up with the pending data layer work. This seems like a good interim solution for now, though. :+1: